### DISCUSSION TITLE
Building wheels for packages with C library dependencies

### DISCUSSION THREAD

**Discussion by dgomes** [home-assistant/architecture#513] [General]
This was triggered by initial discussion in https://github.com/hacs/default/pull/748#issuecomment-775871200

After @ludeeus PR https://github.com/home-assistant/wheels-custom-integrations/pull/279 

I think things are in place to add a new field to the manifest files in https://github.com/home-assistant/wheels-custom-integrations that describe which alpine packages need to be installed in order to successfully create the wheel.

I propose that we add `"apk": list of packages`

Also created an initial PR: https://github.com/home-assistant/wheels-custom-integrations/pull/281

**Comment by frenck** [comment]
There is a rough plan already to add support for packages to the manifest. However, that needs a bit more thought than just the list of APK packages imho (as a matter of fact, been playing with a POC already).

Some considerations/things that might need solving:

- What if not Alpine? (current guess... not our problem)
- Other Alpine repositories?

Especially the last one, is something that might be on our short list for the future. We currently build lots of things in Docker, while we could have provided our own APK repository.

**Reply by dgomes** [reply]
I would propose an MVP approach:
- Just Alpine
- No extra repositories

But please take the front wheel :)

**Reply by frenck** [reply]
Well, the main reason for implementing this in the first place, is providing additional custom APK repositories for our custom builds. Mainly to provide:

- Smaller images
- Short build/release times

Even for an MVP, just a simple `apk` key might be just a bit too simple.

**Reply by dgomes** [reply]
The main reason for an MVPN would be to provide inexistent wheels, but that's just me look into my own https://github.com/home-assistant/wheels-custom-integrations/pull/278 😄 

considering extra repositories, I would use yet another key: `apk_repositories`, that would not interfere with `apk`

**Reply by frenck** [reply]
I agree, but a package would need to be sourced from a specific repository (which has to match the primary release and thus be compatible with the musl version).
